elastic/elastic-agent
# Clarify purpose of `.agent-versions` file

## What does this PR do?

This PR clarifies the purpose of the `.agent-versions` file. It does so by making the following changes:
* Moves and renames the file from `.agent-versions.json` to `testing/integration/testdata/.upgrade-test-agent-versions.yml` and adjust the contents to match.  The reason for using YAML is so we can add a header comment about the contents of the file in the file itself.
* Updates the upgrade integration tests that use this file to reference the new one.
* Updates the commit message in PRs created by automation to keep this file updated.

### Before this PR

Before this PR, we had an `.agent-versions.json` file and it looked like this:

```json
{
  "testVersions": [
    "8.15.2-SNAPSHOT",
    "8.15.1",
    "8.15.0",
    "7.17.25-SNAPSHOT"
  ]
}
```

### After this PR

After this PR, we instead have an `testing/integration/testdata/.upgrade-test-agent-versions.yml` file and it looks like this:

```yml
# This file is generated automatically. Please do not manually edit it.

# The testVersions list in this file specifies Elastic Agent versions to be used as
# the starting (pre-upgrade) or ending (post-upgrade) versions of Elastic Agent in
# upgrade integration tests.

testVersions:
  - 8.15.2-SNAPSHOT
  - 8.15.1
  - 8.15.0
  - 7.17.25-SNAPSHOT
```


## Why is it important?

To make it easier to understand the purpose of the `.agent-versions` file and to prevent accidental manual edits to it.

## How to test this PR locally

1. Create a dummy, valid `testing/integration/testdata/.upgrade-test-agent-versions.yml` file.
2. Run `integration:updateVersions`.
3. Make sure the `testing/integration/testdata/.upgrade-test-agent-versions.yml` file is updated with the expected contents.
